core/src/main/scala/org/apache/spark/deploy/RRunner.scala

```diff
@@ -40,7 +40,13 @@ object RRunner {
 
     // Time to wait for SparkR backend to initialize in seconds
     val backendTimeout = sys.env.getOrElse("SPARKR_BACKEND_TIMEOUT", "120").toInt
-    val rCommand = "Rscript"
+    val rCommand = {
+      var cmd = sys.props.getOrElse("spark.sparkr.r.command", "Rscript")
+      if (sys.props("spark.submit.deployMode") == "client") {
```

Comment:
Yes, we only use RRunner when we launched non-shell jobs (i.e. when we are executing scripts). But R scripts can be executed in either client mode or cluster mode. In client mode, the R script is to be executed on the local host, while in cluster mode, it is to be executed on a selected worker node.